packages/paste-website/src/component-examples/filter/components/FilterPill.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box } from "@twilio-paste/box";
+import { FormPill, type useFormPillState } from "@twilio-paste/form-pill-group";
+import { PlusIcon } from "@twilio-paste/icons/esm/PlusIcon";
+import { Popover, PopoverButton, PopoverContainer, usePopoverState } from "@twilio-paste/popover";
+import React from "react";
+
+import type { FilterMapType, ParticipantsType, selectedFilterProps } from "../types";
+
+const FilterPillView: React.FC<{
+  label: string;
+  selectedType: string | null;
+  selectedValue: selectedFilterProps;
+}> = ({ label, selectedType, selectedValue }) => {
+  if (selectedType === "roomType" && typeof selectedValue === "string") {
+    return (
+      <span>
+        {label}: {selectedValue}
+      </span>
+    );
+  }
+
+  if (selectedType === "participants") {
+    const { min, max } = selectedValue as ParticipantsType;
+
+    return (
+      <span>
+        {label}: {`${min} - ${max}`}
+      </span>
+    );
+  }
+
+  if (selectedType === "dateCompleted" || selectedType === "custom") {
+    const { startDate, endDate } = selectedValue as { startDate: string; endDate: string };
+
+    if (endDate === "") {
+      return (
+        <span>
+          {label}: {startDate}
+        </span>
+      );
+    }
+
+    if (startDate === "") {
+      return (
+        <span>
+          {label}: {endDate}
+        </span>
+      );
+    }
+
+    return (
+      <span>
+        {label}: {startDate} - {endDate}
+      </span>
+    );
+  }
+
+  if (selectedType === "uniqueName") {
+    return (
+      <Box display="flex" alignItems="center">
+        {label}
+      </Box>
+    );
+  }
+
+  return <span>{label}</span>;
+};
+
+export const FilterPill: React.FC<{
+  pill: string;
+  selectedFilters: Record<string, selectedFilterProps>;
+  filterMap: FilterMapType;
+  pillState: ReturnType<typeof useFormPillState>;
+  onDismiss?: () => void;
+  onApply: (type: string, value: selectedFilterProps) => void;
+  onRemove?: () => void;
+}> = ({ pill, selectedFilters, filterMap, pillState, onDismiss, onApply, onRemove }) => {
+  const popover = usePopoverState({ baseId: pill });
+  const isSelected = pill in selectedFilters;
+  const PopoverComponent = filterMap[pill].component;
+  const value = selectedFilters[pill];
+
+  return (
+    <PopoverContainer key={pill} state={popover}>
+      <PopoverButton
+        variant="reset"
+        size="reset"
+        // @ts-expect-error types are wrong
+        borderRadius="borderRadiusPill"
```

Comment:
Was the ticket to expose this prop part of this PR or is there a ticket created?

Reply:
there is a ticket for this https://twilio-productivity.atlassian.net/jira/software/c/projects/DS/boards/25/backlog?selectedIssue=DS-476